### PR TITLE
Bld 29963

### PR DESCRIPTION
**Pull request by tehilaR2**
updates to the async continuity + updating history implementation

### CODE REVIEW

File: SDKSamples/app/src/main/java/com/sdk/samples/topics/AsyncChatContinuity.kt [hunk 1]
```diff
@@ -139,16 +142,56 @@ class AsyncChatForm : Fragment() {
...
+        user_edit.apply {
+            setText(chatViewModel.account?.info?.userInfo?.userId ?: "")
+            filters = arrayOf(InputFilter.LengthFilter(19))
+        }
+        restore_switch.isChecked = chatViewModel.account != null
 
         start_button.setOnClickListener {
 
-            chatViewModel.startChat(ChatData(apiKey_edit.text.toString(), application_edit.text.toString(), restore_switch.isChecked))
+            generateAccount()?.run {
```
Comment (ozbenoved) [inline SDKSamples/app/src/main/java/com/sdk/samples/topics/AsyncChatContinuity.kt]: why not let?

File: SDKSamples/app/src/main/java/com/sdk/samples/topics/History.kt [hunk 2]
```diff
@@ -7,7 +7,7 @@ import com.sdk.samples.R
 import com.sdk.samples.topics.history.HistoryRepository
 import com.sdk.samples.topics.history.RoomHistoryProvider
 
-open class History : BotChat() {
+open class History : BoldChatAsync(){//} BotChat() {
```
Comment (ozbenoved) [inline SDKSamples/app/src/main/java/com/sdk/samples/topics/History.kt]: "//} BotChat() {" can be deleted?

File: SDKSamples/app/src/main/java/com/sdk/samples/topics/AsyncChatContinuity.kt (hunk 1 above)
Comment (tehilaR2) [reply SDKSamples/app/src/main/java/com/sdk/samples/topics/AsyncChatContinuity.kt]: why let?

File: SDKSamples/app/src/main/java/com/sdk/samples/topics/AsyncChatContinuity.kt [hunk 3]
```diff
@@ -91,13 +97,11 @@ open class AsyncChatContinuity : BoldChatAsync() /*[1]*/ {
         chatViewModel = ViewModelProvider(this).get(ChatViewModel::class.java).apply {
 
             onStart.observe(this@AsyncChatContinuity, Observer<ChatData?> { data ->
-                // Start Chat was pressed:
-                data?.apiKey?.takeUnless { it.isBlank() }?.run {
+                //-> Start Chat was pressed:
+                data?.run {
+                    accountRecovery.saveAccount(this)
```
[the remark this answers is not in the document]
Comment (tehilaR2) [reply SDKSamples/app/src/main/java/com/sdk/samples/topics/AsyncChatContinuity.kt]: why let?
i'm usually using let when i want to return a value

File: SDKSamples/app/src/main/java/com/sdk/samples/topics/History.kt (hunk 2 above)
Comment (tehilaR2) [reply SDKSamples/app/src/main/java/com/sdk/samples/topics/History.kt]: no

File: SDKSamples/app/src/main/java/com/sdk/samples/topics/AsyncChatContinuity.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (ozbenoved) [reply SDKSamples/app/src/main/java/com/sdk/samples/topics/AsyncChatContinuity.kt]: ok

File: SDKSamples/app/src/main/java/com/sdk/samples/topics/AsyncChatContinuity.kt (hunk 1 above)
Comment (ozbenoved) [reply SDKSamples/app/src/main/java/com/sdk/samples/topics/AsyncChatContinuity.kt]: ok

File: SDKSamples/app/src/main/java/com/sdk/samples/topics/History.kt (hunk 2 above)
Comment (ozbenoved) [reply SDKSamples/app/src/main/java/com/sdk/samples/topics/History.kt]: ok

**Pull request merged**